app/javascript/pages/browse/MapBrowser.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <section class="MapBrowser">
+    <mapbox
+        access-token="pk.eyJ1IjoibXV0dWFsLWFpZC1hcHAiLCJhIjoiY2tmZTBvd3UwMDBhbTJ4cDlic2JmMWZoaiJ9.rWscBjdl1SMT5N0yekIJYg"
+        :map-options="{
+          style: 'mapbox://styles/mapbox/streets-v11',
+          center: [-96, 37.8],
+          zoom: 4,
+        }"
+        :geolocate-control="{
+          show: true,
+          position: 'top-left',
+        }"
+        @map-load="loaded"
+        @map-zoomend="zoomend"
+        @map-click:points="clicked"
+        @geolocate-error="geolocateError"
+        @geolocate-geolocate="geolocate"
+      />
+  </section>
+</template>
+
+<script>
+import Mapbox from 'mapbox-gl-vue'
+
+export default {
+  components: { Mapbox },
+  methods: {
+    loaded(map) {
+      map.addLayer({
+        id: 'points',
+        type: 'symbol',
+        source: {
+          type: 'geojson',
+        },
+        layout: {
+          'icon-image': '{icon}-15',
+          'text-field': '{title}',
+          'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
+          'text-offset': [0, 0.6],
+          'text-anchor': 'top',
+        },
+      })
+    },
+    zoomend(map, e) {
+      console.log('Map zoomed')
+    },
+    clicked(map, e) {
+      const title = e.features[0].properties.title
+      console.log(title)
+    },
+    geolocateError(control, positionError) {
+      console.log(positionError)
+    },
+    geolocate(control, position) {
+      console.log(
+        `User position: ${position.coords.latitude}, ${position.coords.longitude}`
+      )
+    },
+  },
+}
+</script>
+
+<style>
+  #map {
+    width: 100%;
+    height: 500px;
+  }
+</style>
```

Comment:
This is a great location for these styles. I don't see the `#map` element in the markup here, i assume that brought in by the `mapbox` element?

Reply:
yes, it's referenced from the mapbox module